docs/docs/genai/eval-monitor/scorers/llm-judge/agentic-overview.mdx

```diff
@@ -0,0 +1,203 @@
+- **Debugging**: When failures occur and you need to understand root causes
+- **Optimization**: When identifying performance bottlenecks or inefficient patterns
+- **Refinement**: When improving agent reasoning and decision-making logic
+
+The higher cost and slower execution are justified by the deep insights you gain during these critical development phases, and the quick setup time means you can start evaluating immediately.
+
+### Switch to Field-Based Judges for Production
+
+Field-Based Judges become essential as you approach production:
+
+- **Pre-deployment validation**: Ensure outputs meet quality standards
+- **Regression testing**: Verify that changes don't break existing functionality
+- **Production monitoring**: Continuously assess output quality at scale
+- **SLA compliance**: Verify responses meet defined expectations
+- **A/B testing**: Compare different model versions based on output quality
+
+The fast execution and lower cost make field-based judges ideal for high-volume production evaluation.
+
+## Creating an Agent-as-a-Judge
+
+To create an Agent-as-a-Judge, simply include `{{ trace }}` in your instructions:
+
+```python
+from mlflow.genai.judges import make_judge
+import mlflow
+import time
+
+performance_judge = make_judge(
+    name="performance_analyzer",
+    instructions=(
+        "Analyze the {{ trace }} for performance issues.\n\n"
+        "Check for:\n"
+        "- Operations taking longer than 2 seconds\n"
+        "- Redundant API calls or database queries\n"
+        "- Inefficient data processing patterns\n"
+        "- Proper use of caching mechanisms\n\n"
+        "Rate as: 'optimal', 'acceptable', or 'needs_improvement'"
+    ),
+    model="anthropic:/claude-3-opus-20240229",
+)
+
+
+@mlflow.trace
+def slow_data_processor(query: str):
+    """Example application with performance bottlenecks."""
+    with mlflow.start_span("fetch_data") as span:
+        time.sleep(2.5)
+        span.set_inputs({"query": query})
+        span.set_outputs({"data": ["item1", "item2", "item3"]})
+
+    with mlflow.start_span("process_data") as span:
+        for i in range(3):
+            with mlflow.start_span(f"redundant_api_call_{i}"):
+                time.sleep(0.5)
+        span.set_outputs({"processed": "results"})
+
+    return "Processing complete"
+
+
+result = slow_data_processor("SELECT * FROM users")
+trace_id = mlflow.get_last_active_trace_id()
+trace = mlflow.get_trace(trace_id)
+
+feedback = performance_judge(trace=trace)
+
+print(f"Performance Rating: {feedback.value}")
+print(f"Analysis: {feedback.rationale}")
+```
+
+This example creates an Agent-as-a-Judge that analyzes performance issues, then runs it against a sample application with intentional bottlenecks:
+
+- A database query taking 2.5 seconds (exceeding the 2-second threshold)
+- Three redundant API calls that could be optimized
+
+The judge will output something like:
```

Comment:
@BenWilson2 Can we tell users ho to turn on debug logs to view the agent-as-judge tool calls?